agent/kvs_endpoint_test.go

```diff
@@ -593,11 +593,16 @@ func TestValidateKVKey(t *testing.T) {
 		{"valid key with comma", "foo,bar", false},
 		{"valid key with slash", "foo/bar", false},
 		{"valid key with numbers", "foo123", false},
+		{"valid key with equals", "foo=bar", false},
+		{"valid key with ampersand", "foo&bar", false},
+		{"valid key with question mark", "foo?bar", false},
+		{"valid key with plus", "foo+bar", false},
+		{"valid key with asterisk", "foo*bar", false},
+		{"valid key with percent", "foo%bar", false},
+		{"valid key with C-cedilla", "fooçbar", false},
 		{"empty key", "", true},
 		// Invalid
 		{"invalid key with space", "foo bar", true},
-		{"invalid key with star", "foo*bar", true},
-		{"invalid key with percent", "foo%bar", true},
 		{"invalid key with colon", "foo:bar", true},
 		{"invalid key with semicolon", "foo;bar", true},
 		{"malicious key with path traversal", "../../etc/passwd", true},
```

Comment:
Adding an extra test to confirm that a URL encoded `/` won't work since we'll accept `%` now in the regex and `/` gives `%2F` when URL encoded.
```suggestion
		{"malicious key with path traversal", "../../etc/passwd", true},
		{"malicious key with path traversal with URL encoded /", "..%2F..%2Fetc%2Fpasswd", true},
```